keras_nlp/samplers/beam_sampler.py

```diff
@@ -213,21 +211,19 @@ def gather_beams(x):
         all_log_probs = unflatten_beams(log_probs)
 
         if self.return_all_beams:
-            sorted_indices = tf.argsort(
-                all_log_probs, axis=-1, direction="DESCENDING"
-            )
-            sorted_log_probs = tf.gather(
-                all_log_probs, sorted_indices, axis=-1, batch_dims=1
+            sorted_indices = ops.argsort(all_log_probs, axis=-1)
```

Comment:
We don't have option to set direction in argsort. Because of this couple of tests fail in local with assertion error.

The possible solution could be to reverse the array `sorted_indices` along axis=-1?

Reply:
ops.argsort(-all_log_probs, axis=-1) should do the trick

Reply:
Great! This fixes the failing test

Reply:
You need the minus sign in front so that it is sorted in descending order

Reply:
Flip does the job here. 

> Great! This fixes the failing test for tf backend

I was referring to ops.flip here 😂. Probably you commented at the same time.

Reply:
:o. Missed the `ops.flip` line